linkml/generators/sqltablegen.py

```diff
@@ -238,73 +239,95 @@ def _sql_name(sn: str):
-                    logger.info(
-                        f"WARNING: RANGE EXCEEDS MAXIMUM ORACLE VARCHAR LENGTH, \
-                        CLOB TYPE WILL BE RETURNED: {range} for {slot.name} = {slot.range}"
-                    )
-                    return Text()
-                return VARCHAR2(string_length)
-            if range in ["str", "string", "String", "VARCHAR2", "VARCHAR"]:
-                # If a regex is not detected, string type data should still be represented as a VARCHAR2
-                return VARCHAR2(self.default_length_oracle)
-        if range in schema.classes:
+            schema = self.schema
+
+        sv = SchemaView(schema)
+        if slot_range in sv.all_classes():
             # FK type should be the same as the identifier of the foreign key
-            fk = SchemaView(schema).get_identifier_slot(range)
+            fk = sv.get_identifier_slot(slot_range)
             if fk:
-                return self.get_sql_range(fk, schema)
-            else:
-                return Text()
-        if range in schema.enums:
-            e = schema.enums[range]
+                return self.get_sql_range(fk, sv.schema)
+            return Text()
+
+        if slot_range in sv.all_enums():
+            e = sv.all_enums()[slot_range]
             if e.permissible_values is not None:
                 vs = [str(v) for v in e.permissible_values]
                 return Enum(name=e.name, *vs)
-        if range in METAMODEL_TYPE_TO_BASE:
-            range_base = METAMODEL_TYPE_TO_BASE[range]
-        elif range in schema.types:
-            range_base = schema.types[range].base
-        elif range is None:
-            return Text()
+
+        if slot_range in METAMODEL_TYPE_TO_BASE:
+            range_base = METAMODEL_TYPE_TO_BASE[slot_range]
+        elif slot_range in sv.all_types():
+            range_base = sv.all_types()[slot_range].base
         else:
-            logger.error(f"Unknown range: {range} for {slot.name} = {slot.range}")
+            logger.error(f"Unknown range: {slot_range} for {slot.name} = {slot.range}")
             return Text()
+
         if range_base in RANGEMAP:
             return RANGEMAP[range_base]
-        else:
-            logger.error(f"UNKNOWN range base: {range_base} for {slot.name} = {slot.range}")
-            return Text()
+
+        logger.error(f"Unknown range base: {range_base} for {slot.name} = {slot.range}")
+        return Text()
 
     @staticmethod
-    def get_foreign_key(cn: str, sv: SchemaView) -> str:
-        pk = sv.get_identifier_slot(cn)
-        # TODO: move this to SV
+    def get_id_or_key(cn: str, sv: SchemaView) -> str:
+        """Given a named class, retrieve the identifier or key slot."""
+        pk = sv.get_identifier_slot(cn, use_key=True)
         if pk is None:
-            for sn in sv.class_slots(cn):
-                s = sv.induced_slot(sn, cn)
-                if s.key:
-                    pk = s
-                    break
```

Comment:
this functionality is now in SchemaView so this function can be simplified considerably